src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/INodeWithDebugInfo.cs

```diff
@@ -19,12 +23,37 @@ public DebugLocInfo(int nativeOffset, string fileName, int lineNumber, int colNu
             ColNumber = colNumber;
         }
     }
-
+    
     public interface INodeWithDebugInfo
     {
         DebugLocInfo[] DebugLocInfos
         {
             get;
         }
+
+        DebugVarInfo[] DebugVarInfos
+        {
+            get;
+        }
+    }
+
+     
+    public struct DebugVarInfo
+    {
+        public readonly uint VarNumber; // For parameters, this is the parameter index.
+                                        // For local variables, this is the slot number.
+        public readonly string Name;
+        public readonly uint TypeIndex;
+        public readonly bool IsParam;
+        public List<NativeVarInfo> Ranges;
```

Comment:
Would `ArrayBuilder` make sense here? Seems like it would fit the purpose and it's more lightweight.